checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,353 @@
+ * It will check whether Alluxio classes and Alluxio filesystem can be recognized
+ * in Hadoop task nodes.
+ */
+public class MapReduceIntegrationChecker {
+  private static final Logger LOG = LoggerFactory.getLogger(MapReduceIntegrationChecker.class);
+
+  private static final String FAIL_TO_FIND_CLASS_MESSAGE = "Please distribute "
+      + "the Alluxio client jar on the classpath of the application across different nodes.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hadoop-MapReduce-on-Alluxio.html\n";
+  private static final String FAIL_TO_FIND_FS_MESSAGE = "Please check the fs.alluxio.impl "
+      + "and fs.AbstractFileSystem.alluxio.impl property "
+      + "in core-site.xml file of your Hadoop installation.\n\n"
+      + "For details, please refer to: "
+      + "https://www.alluxio.org/docs/master/en/Running-Hadoop-MapReduce-on-Alluxio.html\n";
+  private static final String TEST_FAILED_MESSAGE = "***** Integration test failed. *****\n";
+  private static final String TEST_PASSED_MESSAGE = "***** Integration test passed. *****\n";
+
+  private static Path sInputFilePath;
+  private static Path sOutputFilePath;
+  private static FileSystem sFileSystem;
+
+  /** The MapReduce task nodes performIntegrationChecks results. */
+  protected enum Status {
+    FAIL_TO_FIND_CLASS, // MapReduce task nodes cannot recognize Alluxio classes
+    FAIL_TO_FIND_FS, // MapReduce task nodes cannot recognize Alluxio filesystem
+    SUCCESS;
+  }
+
+  /**
+   * In each mapper node, we will check whether this node can recognize
+   * Alluxio classes and filesystem.
+   */
+  protected static class CheckStatusMapper extends Mapper<Object, Text, Text, Text> {
+    /**
+     * Gets the Status and IP address of each mapper task node.
+     */
+    protected void map(Object key, Text value, Context context)
+        throws IOException, InterruptedException {
+      context.write(new Text(performIntegrationChecks().toString()), new Text(getAddress()));
+    }
+  }
+
+  /**
+   * In each reducer node, we will combine the IP addresses that have the same Status.
+   */
+  protected static class MergeStatusReducer extends Reducer<Text, Text, Text, Text> {
+    private Text mMergedAddress = new Text();
+
+    /**
+     * Merges the IP addresses of same Status.
+     */
+    protected void reduce(Text key, Iterable<Text> values, Context context)
+        throws IOException, InterruptedException {
+      String mergedAddress = "";
+      for (Text val : values) {
+        String nodeAddress = val.toString();
+        if (!mergedAddress.contains(nodeAddress)) {
+          mergedAddress = mergedAddress + " " + nodeAddress;
+        }
+      }
+      mMergedAddress.set(mergedAddress);
+      context.write(key, mMergedAddress);
+    }
+  }
+
+  /**
+   * @return if this current task node can recognize Alluxio classes and filesystem
+   */
+  private static Status performIntegrationChecks() {
```

Comment:
We do similar checks across all integration checkers. Could we refactor so that there is a single integration-checking utility shared by each checker?